packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -176,6 +176,16 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
         `'${definition.name.value}' is a reserved type name and currently in use within the default schema element.`,
       );
     }
+
+    if (!isDynamoDB) {
+      const containsPrimaryKey = definition.fields?.some((field) => {
+        return field?.directives?.some((directive) => directive.name.value === 'primaryKey');
+      });
+      if (!containsPrimaryKey) {
+        throw new InvalidDirectiveError(`RDS model "${definition.name.value}" must contain a primary key field`);
+      }
+    }
```

Comment:
Instead of pivoting on !isDynamoDB, can we have the generator per-platform have platform-specific `validate` methods which we invoke?

Reply:
That's a good idea. We only have platform specific generators now, will take this as future enhancement. Thanks.